testing/test_assertion.py

```diff
@@ -981,6 +981,36 @@ class SimpleDataObjectTwo:
         assert lines is None
 
 
+class TestAssert_reprcompare_namedtuple:
+    def test_namedtuple(self) -> None:
```

Comment:
Can you also add some tests with objects that have `_fields`, are tuples, but are not NamedTuples?

Also some tests were `_fields` doesn't match `__getattr__` and ` len(type(o)_fields) != len(o)`